src/Analyzer.JsonRuleEngine/JsonRuleEngine.cs

```diff
@@ -21,62 +21,98 @@ public class JsonRuleEngine : IRuleEngine
         /// <param name="context">The <see cref="TemplateContext"/> being evaluated.</param>
         /// <returns>An <see cref="ILineNumberResolver"/> to resolve line numbers for the given template context.</returns>
-        public delegate ILineNumberResolver BuildIJsonLineNumberResolver(TemplateContext context);
+        public delegate ILineNumberResolver BuildILineNumberResolver(TemplateContext context);
 
-        private readonly BuildIJsonLineNumberResolver BuildLineNumberResolver;
+        private readonly BuildILineNumberResolver BuildLineNumberResolver;
+        internal readonly IReadOnlyList<RuleDefinition> RuleDefinitions;
+
+        /// <summary>
+        /// Private constructor to enforce use of <see cref="JsonRuleEngine.Create(string, BuildILineNumberResolver)"/> for creating new instances.
+        /// </summary>
+        private JsonRuleEngine(List<RuleDefinition> rules, BuildILineNumberResolver jsonLineNumberResolverBuilder)
+        {
+            this.RuleDefinitions = rules;
+            this.BuildLineNumberResolver = jsonLineNumberResolverBuilder;
+        }
 
         /// <summary>
         /// Creates an instance of <see cref="JsonRuleEngine"/>.
         /// </summary>
+        /// <param name="rawRuleDefinitions">The raw JSON rules to evaluate a template with.</param>
         /// <param name="jsonLineNumberResolverBuilder">A builder to create an <see cref="ILineNumberResolver"/> for mapping JSON paths from a
         /// processed template to the line number of the equivalent location in the original template.</param>
-        public JsonRuleEngine(BuildIJsonLineNumberResolver jsonLineNumberResolverBuilder)
+        public static JsonRuleEngine Create(string rawRuleDefinitions, BuildILineNumberResolver jsonLineNumberResolverBuilder)
         {
-            this.BuildLineNumberResolver = jsonLineNumberResolverBuilder ?? throw new ArgumentNullException(nameof(jsonLineNumberResolverBuilder));
+            if (jsonLineNumberResolverBuilder == null) throw new ArgumentNullException(nameof(jsonLineNumberResolverBuilder));
+
+            try
+            {
+                return new JsonRuleEngine(ParseRuleDefinitions(rawRuleDefinitions), jsonLineNumberResolverBuilder);
+            }
+            catch (Exception e)
+            {
+                throw new JsonRuleEngineException($"Failed to parse rules.", e);
+            }
         }
 
         /// <summary>
         /// Evaluates a template using rules defined in JSON.
         /// </summary>
         /// <param name="templateContext">The template context to evaluate.</param>
-        /// <param name="ruleDefinitions">The JSON rules to evaluate the template with.</param>
         /// <returns>The results of the rules against the template.</returns>
-        public IEnumerable<IEvaluation> EvaluateRules(TemplateContext templateContext, string ruleDefinitions)
+        public IEnumerable<IEvaluation> EvaluateTemplate(TemplateContext templateContext)
```

Comment:
Should this be `AnalyzeTemplate` instead?

Reply:
This would obviously be a change in IRuleEngine

Reply:
I wasn't sure - I changed from `Rules` to `Template` since the rules aren't passed in this function anymore.  Whether this should be `AnalyzeTemplate` or `EvaluateTemplate` wasn't as clear to me.  (I changed the function in `TemplateAnalyzer` because an analyzer analyzes things, so it made sense.)

I'm happy to rename if there's a strong general feeling it should change (to be consistent, for ex.).

Reply:
I like the idea of keeping the language consistent, but I guess it's not too big of a deal.

Reply:
Renamed them.